src/main/java/io/quarkus/search/app/SearchService.java

```diff
@@ -85,12 +85,12 @@ public SearchResult<GuideSearchHit> search(@RestQuery @DefaultValue(QuarkusVersi
     private ElasticsearchSearchResult<GuideSearchHit> performSearch(String version, List<String> categories, String q,
             String origin, Language language, String highlightCssClass, int page, int contentSnippets,
             int contentSnippetsLength, SearchSession session) {
-        return session.search(Guide.class)
+        return session.search(Guide__.INDEX.scope(session))
                 .extension(ElasticsearchExtension.get())
                 .select(f -> f.composite().from(
                         f.id(),
-                        f.field("type"),
-                        f.field("origin"),
+                        f.field(Guide__.INDEX.type),
+                        f.field(Guide__.INDEX.origin),
                         f.highlight(language.addSuffix("title")).highlighter("highlighter_title_or_summary").optional(),
```

Comment:
Could `language.addSuffix("title")` possibly be replaced with `language.select(Guide__.INDEX.title)`, assuming we turn `title` into an object field (use `title.fr`, `title.en`, ... instead of `title_fr`, `title_en`, ...)?

Reply:
Ah right, I forgot custom binders weren't supported yet. Sorry.